content/tutorials/token-authentication.textile

```diff
@@ -346,7 +457,38 @@ blang[ruby].
     end
   ```
 
-Both the logged in and anonymous HTML panels will currently show on the HTML page, let's add quickly hide or show them based on the cookie set above. Add the following just before the final @</script>@ tag in @index.html@
+blang[php].
+  Now we need to enable the login and logout functionality in the web server. In order to keep this tutorial simple, when a user logs in we'll set a cookie, and when they log out we'll clear the cookie.
+
+  We'll create two scripts that will serve as our login and logout functionality.
+
+  First create @login.php@:
+
+  ```[php]
+    <?php
+    if (isset($_REQUEST['username'])) {
+      setcookie('username', $_REQUEST['username']);
+      header('Location: /', true, 302);
+    } else {
+      http_response_code(500);
+      echo "Username is required to login";
+    }
+  ```
+
+  Second script will serve as logout, create @logout.php@:
+
+  ```[php]
+    <?php
+    setcookie('username', null, -1, '/');
+    header('Location: /', true, 302);
+  ```
+
+
+blang[nodejs,ruby].
+  Both the logged in and anonymous HTML panels will currently show on the HTML page, let's add quickly hide or show them based on the cookie set above. Add the following just before the final @</script>@ tag in @index.html@
+blang[php].
```

Comment:
White space issue here again I believe.  Please confirm the output is as expected